src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -1509,44 +1512,46 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
     /** Get type info associated with symbol at current phase, after
      *  ensuring that symbol is initialized (i.e. type is completed).
      */
-    def info: Type = try {
-      var cnt = 0
-      while (validTo == NoPeriod) {
-        assert(infos ne null, this.name)
-        assert(infos.prev eq null, this.name)
-        val tp = infos.info
-
-        if ((_rawflags & LOCKED) != 0L) { // rolled out once for performance
-          lock {
-            setInfo(ErrorType)
-            throw CyclicReference(this, tp)
+    def info: Type = Parallel.synchronizeAccess(SymbolLock) {
```

Comment:
We need to make sure this and tpe constructors synchronized on `synchronizeSymbolsAccess` shouldn't use the same lock.

Pros of the same lock:
- safer
- easier to reason about

Cons:
- maybe slower

If you are 100% sure they are independent please add simple comment explaining why.

Reply:
We need to discuss this one